manticore/core/cpu/abstractcpu.py

```diff
@@ -387,7 +458,9 @@ def read_register(self, register):
         :return: register value
         :rtype: int or long or Expression
         '''
-        return self._regfile.read(register)
+        value = self._regfile.read(register)
+        self.will_read_register(register, value)
```

Comment:
Sort of inaccurate, `will_read_register` should execute before the actual `read`. (i.e. if `read` raises, the signal won't fire)